// ngx-init/esm2015/lib/ngx-init.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
export class NgxInitDirective {
    /**
     * @param {?} templateRef
     * @param {?} viewContainer
     */
    constructor(templateRef, viewContainer) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set ngxInit(val) {
        this.viewContainer.clear();
        this.viewContainer.createEmbeddedView(this.templateRef, { ngxInit: val });
    }
}
NgxInitDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngxInit]',
            },] },
];
/** @nocollapse */
NgxInitDirective.ctorParameters = () => [
    { type: TemplateRef, },
    { type: ViewContainerRef, },
];
NgxInitDirective.propDecorators = {
    "ngxInit": [{ type: Input },],
};
function NgxInitDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxInitDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxInitDirective.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    NgxInitDirective.propDecorators;
    /** @type {?} */
    NgxInitDirective.prototype.templateRef;
    /** @type {?} */
    NgxInitDirective.prototype.viewContainer;
}
