// ngx-init/bundles/ngx-init.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ngx-init', ['exports', '@angular/core'], factory) :
    (factory((global['ngx-init'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxInitDirective = (function () {
        function NgxInitDirective(templateRef, viewContainer) {
            this.templateRef = templateRef;
            this.viewContainer = viewContainer;
        }
        Object.defineProperty(NgxInitDirective.prototype, "ngxInit", {
            set: /**
             * @param {?} val
             * @return {?}
             */ function (val) {
                this.viewContainer.clear();
                this.viewContainer.createEmbeddedView(this.templateRef, { ngxInit: val });
            },
            enumerable: true,
            configurable: true
        });
        NgxInitDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[ngxInit]',
                    },] },
        ];
        /** @nocollapse */
        NgxInitDirective.ctorParameters = function () {
            return [
                { type: core.TemplateRef, },
                { type: core.ViewContainerRef, },
            ];
        };
        NgxInitDirective.propDecorators = {
            "ngxInit": [{ type: core.Input },],
        };
        return NgxInitDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxInitModule = (function () {
        function NgxInitModule() {
        }
        NgxInitModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [],
                        declarations: [NgxInitDirective],
                        exports: [NgxInitDirective]
                    },] },
        ];
        return NgxInitModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.NgxInitDirective = NgxInitDirective;
    exports.NgxInitModule = NgxInitModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
