// ngx-init/fesm5/ngx-init.js
import { Directive, Input, TemplateRef, ViewContainerRef, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxInitDirective = /** @class */ (function () {
    function NgxInitDirective(templateRef, viewContainer) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    Object.defineProperty(NgxInitDirective.prototype, "ngxInit", {
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this.viewContainer.clear();
            this.viewContainer.createEmbeddedView(this.templateRef, { ngxInit: val });
        },
        enumerable: true,
        configurable: true
    });
    NgxInitDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[ngxInit]',
                },] },
    ];
    /** @nocollapse */
    NgxInitDirective.ctorParameters = function () { return [
        { type: TemplateRef, },
        { type: ViewContainerRef, },
    ]; };
    NgxInitDirective.propDecorators = {
        "ngxInit": [{ type: Input },],
    };
    return NgxInitDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxInitModule = /** @class */ (function () {
    function NgxInitModule() {
    }
    NgxInitModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [NgxInitDirective],
                    exports: [NgxInitDirective]
                },] },
    ];
    return NgxInitModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxInitDirective, NgxInitModule };
