// ngx-init/fesm2015/ngx-init.js
import { Directive, Input, TemplateRef, ViewContainerRef, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxInitDirective {
    /**
     * @param {?} templateRef
     * @param {?} viewContainer
     */
    constructor(templateRef, viewContainer) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set ngxInit(val) {
        this.viewContainer.clear();
        this.viewContainer.createEmbeddedView(this.templateRef, { ngxInit: val });
    }
}
NgxInitDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngxInit]',
            },] },
];
/** @nocollapse */
NgxInitDirective.ctorParameters = () => [
    { type: TemplateRef, },
    { type: ViewContainerRef, },
];
NgxInitDirective.propDecorators = {
    "ngxInit": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxInitModule {
}
NgxInitModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [NgxInitDirective],
                exports: [NgxInitDirective]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxInitDirective, NgxInitModule };
